multiqc/modules/checkatlas/checkatlas.py

```diff
@@ -0,0 +1,544 @@
+    list_qc_mito.sort(reverse=True)
+
+    for i in range(1, len(list_qc_counts)):
+        data_qc_counts[math.log10(i)] = list_qc_counts[i]
+    for i in range(1, len(list_qc_genes)):
+        data_qc_genes[math.log10(i)] = list_qc_genes[i]
+    for i in range(1, len(list_qc_mito)):
+        data_qc_mito[math.log10(i)] = list_qc_mito[i]
+
+    return data_qc_counts, data_qc_genes, data_qc_mito
+
+
+def parse_firstline_table_logs(f):
+    """
+    Parse only header and first line of logs which are .tsv files
+    Ex: _checkatlas.tsv, adata_checkatlas.tsv
+    :param f:
+    :return:
+    """
+    data = {}
+    lines = f.splitlines()
+    headers = lines[0].split("\t")
+    for i in range(1, len(lines)):
+        line = lines[i].split("\t")
+        for j in range(0, len(line)):
+            data[headers[j]] = line[j]
+    return data
+
+
+def parse_metric_logs(f):
+    """
+    Parse all lines of logs which are .tsv files
+    Ex: _checkatlas.tsv, adata_checkatlas.tsv
+    :param f:
+    :return:
+    """
+    data = {}
+    lines = f.splitlines()
+    headers = lines[0].split("\t")
+    for i in range(1, len(lines)):
+        line = lines[i].split("\t")
+        line_dict = {}
+        for j in range(1, len(line)):
+            line_dict[headers[j]] = line[j]
+        data[line[0]] = line_dict
+    return data
+
+
+def create_img_html_content(type_viz, data):
+    html_content = """
+            <div class="tab">\n"""
+    counter = 0
+    tablinks = type_viz + "_tablinks"
+    tabcontent = type_viz + "_tabcontent"
+    for key, value in data.items():
+        class_tab = tablinks
+        if counter == 0:
+            class_tab = tablinks + " active"
+        html_content += add_selection_img_button(class_tab, type_viz, key, tablinks, tabcontent)
+        counter += 1
+    html_content += """</div>"""
+
+    counter = 0
+    for key, value in data.items():
+        path_fig = os.path.join(config.data_dir_name, FIG_PATH, value)
+        style = "display: none;"
+        if counter == 0:
+            style = "display: block;"
+        html_content += add_div_img(path_fig, type_viz, key, style, tabcontent)
+        counter += 1
+
+    html_content += openpng_html_script
+    return html_content
+
+
+def add_selection_img_button(class_tab, type_viz, key, tablinks, tabcontent):
+    return (
+        """  <button class=\""""
+        + class_tab
+        + """\" onclick="openPNG(event, '"""
+        + type_viz
+        + "_"
+        + key
+        + """',
+                            '"""
+        + tablinks
+        + "','"
+        + tabcontent
+        + """')">"""
+        + key
+        + """</button>\n"""
+    )
+
+
+def add_div_img(path_fig, type_viz, key, style, tabcontent):
+    return (
+        """<div id=\""""
+        + type_viz
+        + "_"
+        + key
+        + """\" class=\""""
+        + tabcontent
+        + """\" style =\""""
+        + style
+        + """\"><img src=\""""
+        + path_fig
```

Comment:
This doesn't work for MultiQC. Reports are entirely self-enclosed, without any external dependencies. They can be moved around as a single file, attached to emails etc and they still work.

Images, when used, are base64-encoded and included within the HTML.